### ISSUE TITLE
[FORMATTING] Formatting for UNNEST with ARRAY<STRUCT<...>> breaks the SQL - BigQuery

### ISSUE THREAD

**Issue by m-ahmadkhan** [sql-formatter-org/sql-formatter#279]
**Input data**
UNNEST and STRUCT example given here (https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#unnest_and_structs)

```sql
SELECT *
FROM UNNEST(ARRAY<STRUCT<x INT64, y STRING>>[(1, 'foo'), (3, 'bar')]);
```

**Expected Output**

```sql
SELECT
  *
FROM
  UNNEST(
    ARRAY<STRUCT<x INT64, y STRING>>[(1, 'foo'), (3, 'bar')]
  );
```

**Actual Output**

```sql
SELECT
  *
FROM
UNNEST
  (
    ARRAY<STRUCT<xINT64,ySTRING>>[(1, 'foo'), (3, 'bar')]
  );
```

There are two issues with the output:

1. UNNEST shouldn't be formatted on the same indent level as FROM.
2. It's breaking the SQL by removing the spaces for the definition of the struct `STRUCT<xINT64,ySTRING>`

**Usage**

- How are you calling / using the library?
   - Tested by writing a test in `bigquery.test.ts`:
```
it('formats unnest with array<struct> correctly', () => {
    expect(
      format(`
      SELECT *
      FROM UNNEST(ARRAY<STRUCT<x INT64, y STRING>>[(1, 'foo'), (3, 'bar')]);
      `)
    ).toBe(dedent`
      SELECT
        *
      FROM
        UNNEST(
          ARRAY<STRUCT<x INT64, y STRING>>[(1, 'foo'), (3, 'bar')]
        );
    `);
  });
```
- What SQL language(s) does this apply to?
   - BigQuery
- Which SQL Formatter version are you using?
   - v8.0.1

**Comment by nene** [comment]
Thanks for reporting.

Fixed UNNEST formatting.

Will look into fixing STRUCT formatting as well, though I suspect it'll be trickier to correct, as currently the formatting of these type parameters is quite a hack.

**Comment by nene** [comment]
Found a fix for that other problem too and released both in 8.0.2
